src/Tizen.NUI.WindowSystem/src/public/EdgeDragData.cs

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright(c) 2023 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using static Tizen.NUI.WindowSystem.Interop.InputGesture;
+
+namespace Tizen.NUI.WindowSystem
+{
+    /// <summary>
+    /// Represents a wrapper class for a unmanaged gesture data handle.
+    /// </summary>
+    public class EdgeDragData : IDisposable
+    {
+        private InputGesture _gesture;
+        internal SafeGestureHandle _handle;
+        private bool disposed = false;
+
+        private static event EventHandler<EdgeDragEventArgs> _eventHandler;
+        private static EdgeDragCb _delegate;
+
+        /// <summary>
+        /// Creates a new EdgeDragData.
+        /// </summary>
+        public EdgeDragData(InputGesture gesture, int fingers, GestureEdge edge)
+        {
+            if (gesture == null)
+            {
+                throw new ArgumentNullException(nameof(gesture));
+            }
+            if (gesture.GetNativeHandle().IsInvalid)
+            {
+                throw new ArgumentException("InputGesture is not initialized.");
+            }
+            _gesture = gesture;
+            _handle = EdgeDragNew(_gesture.GetNativeHandle(), fingers, (int)edge);
+            if (_handle.IsInvalid)
+            {
+                int err = Tizen.Internals.Errors.ErrorFacts.GetLastResult();
+                _gesture.ErrorCodeThrow((Interop.InputGesture.ErrorCode)err);
+            }
+            Log.Debug(LogTag, "EdgeDragData Created");
+        }
+
+        /// <summary>
+        /// Sets a specific size of edge for edge drag gesture
+        /// </summary>
+        /// <param name="edgeSize"> The enum of gesture edge size </param>
+        /// <param name="startPoint"> The start point of edge area </param>
+        /// <param name="endPoint"> The end point of edge area</param>
+        public void SetEdgeSize(GestureEdgeSize edgeSize, int startPoint, int endPoint)
+        {
+            Interop.InputGesture.ErrorCode res = Interop.InputGesture.EdgeDragSizeSet(_handle, (int)edgeSize, startPoint, endPoint);
+            _gesture.ErrorCodeThrow(res);
+            Log.Debug(LogTag, "EdgeDragData.SetEdgeSize");
+        }
+
+        /// <summary>
+        /// Grabs a global gesture
+        /// </summary>
+        public void GestureGrab()
```

Comment:
Because methods are the means of taking action, the design guidelines require that method names be verbs or verb phrases. Following this guideline also serves to distinguish method names from property and type names, which are noun or adjective phrases.

✔️ DO give methods names that are verbs or verb phrases.

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members